pkg/extensions/search/resolver.go

```diff
@@ -1193,41 +1193,33 @@ func getImageList(ctx context.Context, repo string, repoDB repodb.RepoDB, cveInf
 	}, nil
 }
 
-func getReferrers(store storage.ImageStore, repoName string, digest string, artifactTypes []string, log log.Logger) (
-	[]*gql_generated.Referrer, error,
-) {
-	results := make([]*gql_generated.Referrer, 0)
-
-	index, err := store.GetReferrers(repoName, godigest.Digest(digest), artifactTypes)
-	if err != nil {
-		log.Error().Err(err).Msg("error extracting referrers list")
+func getReferrers(repoDB repodb.RepoDB, repo string, referredDigest string, artifactTypes []string,
+	log log.Logger,
+) ([]*gql_generated.Referrer, error) {
+	refDigest := godigest.Digest(referredDigest)
+	if err := refDigest.Validate(); err != nil {
+		log.Error().Err(err).Msgf("graphql: bad digest string from request '%s'", referredDigest)
 
-		return results, err
+		return []*gql_generated.Referrer{}, errors.Wrapf(err, "graphql: bad digest string from request '%s'",
+			referredDigest)
 	}
 
-	for _, manifest := range index.Manifests {
-		size := int(manifest.Size)
-		digest := manifest.Digest.String()
-		annotations := make([]*gql_generated.Annotation, 0)
-		artifactType := manifest.ArtifactType
-		mediaType := manifest.MediaType
+	referrers, err := repoDB.GetFilteredReferrersInfo(repo, refDigest, artifactTypes)
```

Comment:
This is ok. Maybe in the future pass a parameter of type function containing the filtering logic (similar to how we currently filter tags).

Reply:
We'll refactor later if we need to.